src/System.Private.CoreLib/src/System/MulticastDelegate.cs

```diff
@@ -135,32 +136,32 @@ public override sealed bool Equals(object obj)
         private bool InvocationListEquals(MulticastDelegate d)
         {
             Debug.Assert(d != null && (_invocationList as object[]) != null, "bogus delegate in multicast list comparison");
-            object[] invocationList = _invocationList as object[];
+            object[] invocationList = (_invocationList as object[])!; // Calling code ensures invocationList is object[]
             if (d._invocationCount != _invocationCount)
                 return false;
 
             int invocationCount = (int)_invocationCount;
             for (int i = 0; i < invocationCount; i++)
             {
-                Delegate dd = (Delegate)invocationList[i];
-                object[] dInvocationList = d._invocationList as object[];
+                Delegate dd = (Delegate)invocationList[i]!; // If invocationList is an object[], it always contains Delegate (or MulticastDelegate) objects
```

Comment:
I don't understand why there's a warning here. `invocationList` is `object[]`; as far as the compiler is concerned, it shouldn't contain nulls, right?

Also, in general, rather than adding comments that make claims like this, I'd rather assert, so that the "comment" is actually validated.  In many cases that'll also obviate the need to use `!`, as the compiler respects assertions made in Debug.Assert even though they're compiled away in release.